Gems/Atom/RPI/Code/Source/RPI.Public/Image/ImageTagSystemComponent.cpp

```diff
@@ -0,0 +1,133 @@
+                ;
+            }
+        }
+
+        void ImageTagSystemComponent::GetRequiredServices(ComponentDescriptor::DependencyArrayType& /*required*/)
+        {
+        }
+
+        void ImageTagSystemComponent::GetProvidedServices(ComponentDescriptor::DependencyArrayType& provided)
+        {
+            provided.push_back(AZ_CRC_CE("TextureTagSystemComponent"));
+        }
+
+        void ImageTagSystemComponent::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& /*dependent*/)
+        {
+        }
+
+        void ImageTagSystemComponent::Activate()
+        {
+            ImageTagBus::Handler::BusConnect();
+        }
+
+        void ImageTagSystemComponent::Deactivate()
+        {
+            ImageTagBus::Handler::BusDisconnect();
+        }
+
+        ImageQuality ImageTagSystemComponent::GetQuality(const AZ::Name& imageTag) const
+        {
+            auto it = m_imageTags.find(imageTag);
+            if (it == m_imageTags.end())
+            {
+                AZ_Warning("ImageTagSystemComponent", false, "Image tag %s has not been registered", imageTag.GetCStr());
+                return ImageQualityHighest;
+            }
+
+            return it->second.quality;
+        }
+
+        AZStd::vector<AZ::Name> ImageTagSystemComponent::GetTags() const
+        {
+            AZStd::vector<AZ::Name> tags;
+            for (auto&& [tag, tagData] : m_imageTags)
+            {
+                tags.push_back(tag);
+            }
+
+            AZStd::sort(tags.begin(), tags.end(), [](const AZ::Name& lhs, const AZ::Name& rhs) { return lhs.GetStringView() < rhs.GetStringView(); });
+
+            return tags;
+        }
+
+        void ImageTagSystemComponent::RegisterImageAsset(AZ::Name imageTag, const Data::AssetId& assetId)
+        {
+            auto it = m_imageTags.find(imageTag);
+            if (it == m_imageTags.end())
+            {
+                AZ_Warning("ImageTagSystemComponent", false, "Image tag %s has not been registered", imageTag.GetCStr());
+                return;
+            }
+
+            TagData& tagData = it->second;
+            tagData.registeredImages.insert(assetId);
+        }
+
+        void ImageTagSystemComponent::RegisterTag(AZ::Name imageTag)
+        {
+            AZ_Warning("ImageTagSystemComponent", m_imageTags.find(imageTag) == m_imageTags.end(), "Image tag %s has already been registered", imageTag.GetCStr());
+
+            m_imageTags.emplace(AZStd::move(imageTag), TagData{});
+        }
+
+        void ImageTagSystemComponent::SetQuality(const AZ::Name& imageTag, ImageQuality quality)
+        {
+            auto it = m_imageTags.find(imageTag);
+            if (it == m_imageTags.end())
+            {
+                AZ_Warning("ImageTagSystemComponent", false, "Image tag %s has not been registered", imageTag.GetCStr());
+                return;
+            }
+
+            TagData& tagData = it->second;
+            tagData.quality = quality;
```

Comment:
You may check if the quality is change and only reload assets if the quality is changed.